core/arch/arm/include/kernel/linker.h

```diff
@@ -58,15 +58,15 @@ extern const struct core_mmu_phys_mem __end_phys_mem_map_section;
 extern const struct core_mmu_phys_mem __start_phys_nsec_ddr_section;
 extern const struct core_mmu_phys_mem __end_phys_nsec_ddr_section;
 
-#define VCORE_UNPG_RX_PA	((paddr_t)__vcore_unpg_rx_start)
+#define VCORE_UNPG_RX_PA	((unsigned long)__vcore_unpg_rx_start)
```

Comment:
As these hold physical addresses/sizes, `CFG_CORE_LARGE_PHYS_ADDR=y` requires 64bit fields here. Why doesn't `paddr_t` fit?

Reply:
Static initialization based on addresses of variables seems to the a bit fragile. As soon as there's a 64-bit type on a 32-bit system involved the compiler tends to complain. `unsigned long` should be a good enough compromise for all systems.

Reply:
ok